libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts

```diff
@@ -245,6 +232,106 @@ export const ParentalLeaveForm: Form = buildForm({
             }),
           ],
         }),
+        buildSubSection({
+          id: 'personalAllowanceSubSection',
+          name: m.personalAllowanceName,
+          children: [
+            buildRadioField({
+              id: 'usePersonalAllowance',
+              name: m.usePersonalAllowance,
+              options: [
+                { label: m.yesOptionLabel, value: 'yes' },
+                { label: m.noOptionLabel, value: 'no' },
+              ],
+            }),
+            buildMultiField({
+              id: 'personalAllowance',
+              condition: (answers) => answers.usePersonalAllowance === 'yes',
+              name: m.personalAllowanceName,
+              description: m.personalAllowanceDescription,
+              children: [
+                buildSelectField({
+                  // This should probably be a text input with a format, and type number
+                  name: m.paymentInformationPersonalAllowance,
+                  id: 'personalAllowance.usage',
+                  width: 'half',
+                  options: [
+                    { label: '100%', value: '100' },
+                    { label: '75%', value: '75' },
+                    { label: '50%', value: '50' },
+                    { label: '25%', value: '25' },
```

Comment:
Are values always strings?
Does the schema handle converting this to a number?

Same below

Reply:
So this is one of the things that definitely needs improvement in the application system. React hook form gets all the values from the inputs that are registered on each screen, and the values are almost always of type string. This can be manipulated by the use of `setValue` and casting to the desired type, but that is problematic as well.

Zod, the dataschema library, does not yet support transformation properly, but version 2 which is in beta does. Then we can use the dataschema lib to transform the strings into desired formats when needed and store the answers like that properly. 

I am all ears if you have any ideas how we can improve this, as I agree that this is not good enough